refactor(docs-infra): minor refactorings/style changes

gkalpak opened the pull request:
This PR includes some minor refactorings and style changes as a follow-up to PR #41129.
(The changes were minor enough that didn't warrant blocking the PR.)

mary-poppins commented:
You can preview cdb40a0 at https://pr42396-cdb40a0.ngbuilds.io/.

petebacondarwin commented on aio/src/styles/1-layouts/marketing-layout/_marketing-layout-theme.scss:
```diff
@@ -13,10 +13,6 @@
-
-    .hero-logo {
-      background-image: url(/assets/images/logos/angular/angular.svg);
-    }
```
Can you explain why the image is moved but the `color` of the `#intro` is not?
I assume that the image is moved because it does not change whatever the current theme is?
But it seems to me that this is also true of the `color`?

Either way, it would be great if we could have some comments in this file and the `_marketing-layout.scss` explaining a bit about how all this works.

gkalpak replied:
As discussed "offline", I have added some docs on styling/theming in https://github.com/angular/angular/pull/42396/commits/9b9a2f6eeff9a5c64d57cf0a24c3ce8d4ca17b25.

gkalpak commented:
@petebacondarwin, I have added three new commits with some more refactorings/fixes/docs changes. PTAL 🙏

mary-poppins commented:
You can preview 9b9a2f6 at https://pr42396-9b9a2f6.ngbuilds.io/.

petebacondarwin commented on aio/src/styles/README.md:
```diff
@@ -0,0 +1,80 @@
+
+For each area/component, there is a sub-directory in either `1-layouts/` or `2-modules/`.
+
+Each such sub-directory contains a `<name>.scss` file with styles for the corresponding area/component and may also contains a `<name>-theme.scss` file with styles related to theming.
```
`contains` -> `contain`

petebacondarwin reviewed (approved):
Nice - one typo in the content and one in a commit:

```
fix(docs-infra): improve homepage layout in print mode
In #41129, the `.hero-logo` image what changed from an `<img>` element
                                  ^^^^
```

gkalpak committed:
refactor(docs-infra): remove unused `_typography-theme.scss` file

This commit removes the `_typography-theme.scss` file that is currently
not used. The file contains a single Sass mixin
(`docs-site-typography-theme()`), which is never called.

gkalpak committed:
refactor(docs-infra): minor refactorings/style changes

This commit includes some minor refactorings and style changes as a
follow-up to PR #41129. (The changes were minor enough that didn't
warrant blocking the PR.)

gkalpak committed:
fix(docs-infra): improve homepage layout in print mode

In #41129, the `.hero-logo` image was changed from an `<img>` element to
a `<div>` with an image background. As a result, in print mode (where
browsers often omit backgrounds to save ink) the logo was not present.

This commit fixes this by ensuring the word `Angular` is displayed as a
title instead of the logo in print mode. It also hides the `Get started`
button in print mode, since it doesn't add any value.

Before: ![homepage-print before][1]
After: ![homepage-print after][2]

[1]: https://user-images.githubusercontent.com/8604205/120068867-889e6080-c08b-11eb-9bd5-233d8592f7ec.png
[2]: https://user-images.githubusercontent.com/8604205/120068869-89cf8d80-c08b-11eb-8740-a68a5ef68c34.png

gkalpak committed:
docs(docs-infra): document the implementation of styles (incl. theming) in angular.io

gkalpak committed:
fixup! docs(docs-infra): document the implementation of styles (incl. theming) in angular.io

mary-poppins commented:
You can preview 83a5874 at https://pr42396-83a5874.ngbuilds.io/.

petebacondarwin reviewed (approved):
Reviewed-for: global-docs-approvers
